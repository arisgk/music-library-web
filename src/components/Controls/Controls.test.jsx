/* global describe, beforeAll, it, expect */
import React from 'react';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import { createShallow } from '@material-ui/core/test-utils';
import Wrapper, { Controls } from './index';

Enzyme.configure({ adapter: new Adapter() });

describe('Controls', () => {
  let shallow;
  let songs;
  let onTogglePlay;
  let onPreviousClick;
  let onNextClick;

  beforeAll(() => {
    shallow = createShallow({ untilSelector: Controls });
    songs = [1, 2, 3];
    onTogglePlay = () => {};
    onPreviousClick = () => {};
    onNextClick = () => {};
  });

  it('renders correctly', () => {
    const song = {
      id: 1,
      title: 'Animals',
      url: 'https://some-host.com/animals',
      artist: 'Muse',
      album: 'The 2nd Law',
      albumCover: 'https://some-host.com/muse-the-2nd-law.jpg',
      duration: 263,
    };
    let playing = false;
    const isFirstSong = true;
    const isLastSong = false;

    let wrapper = shallow(
      <Wrapper
        songs={songs}
        song={song}
        playing={playing}
        isFirstSong={isFirstSong}
        isLastSong={isLastSong}
        onTogglePlay={onTogglePlay}
        onPreviousClick={onPreviousClick}
        onNextClick={onNextClick}
      />,
    );
    expect(toJson(wrapper)).toMatchSnapshot();

    playing = true;

    wrapper = shallow(
      <Wrapper
        songs={songs}
        song={song}
        playing={playing}
        isFirstSong={isFirstSong}
        isLastSong={isLastSong}
        onTogglePlay={onTogglePlay}
        onPreviousClick={onPreviousClick}
        onNextClick={onNextClick}
      />,
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
