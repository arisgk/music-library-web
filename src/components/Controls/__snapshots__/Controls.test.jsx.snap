// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls renders correctly 1`] = `
<div>
  <audio
    src="https://some-host.com/animals"
  />
  <div
    className="Controls-controlsContainer-1"
  >
    <div
      className="Controls-songSequenceControlsContainer-2"
    >
      <WithStyles(IconButton)
        aria-label="Previous"
        disabled={true}
        onClick={[Function]}
      >
        <pure(SkipPreviousIcon) />
      </WithStyles(IconButton)>
      <WithStyles(IconButton)
        aria-label="Play/Pause"
        onClick={[Function]}
      >
        <pure(PlayCircleOutlineIcon)
          fontSize="large"
        />
      </WithStyles(IconButton)>
      <WithStyles(IconButton)
        aria-label="Next"
        disabled={false}
        onClick={[Function]}
      >
        <pure(SkipNextIcon) />
      </WithStyles(IconButton)>
    </div>
    <div
      className="Controls-progressContainer-3"
    >
      <WithStyles(LinearProgress)
        classes={
          Object {
            "dashed": undefined,
            "root": "Controls-linearProgressRoot-4",
          }
        }
        value={0}
        variant="determinate"
      />
    </div>
    <MediaQuery
      query="(min-device-width: 768px)"
      values={Object {}}
    >
      <div
        className="Controls-metaContainer-5"
      >
        <img
          className="Controls-cover-9"
          role="presentation"
          src="https://some-host.com/muse-the-2nd-law.jpg"
        />
        <div
          className="Controls-infoContainer-6"
        >
          <h4
            className="Controls-songTitle-7"
          >
            Animals
          </h4>
          <h5
            className="Controls-songArtist-8"
          >
            Muse
          </h5>
        </div>
      </div>
    </MediaQuery>
  </div>
</div>
`;

exports[`Controls renders correctly 2`] = `
<div>
  <audio
    src="https://some-host.com/animals"
  />
  <div
    className="Controls-controlsContainer-1"
  >
    <div
      className="Controls-songSequenceControlsContainer-2"
    >
      <WithStyles(IconButton)
        aria-label="Previous"
        disabled={true}
        onClick={[Function]}
      >
        <pure(SkipPreviousIcon) />
      </WithStyles(IconButton)>
      <WithStyles(IconButton)
        aria-label="Play/Pause"
        onClick={[Function]}
      >
        <pure(PauseCircleOutlineIcon)
          fontSize="large"
        />
      </WithStyles(IconButton)>
      <WithStyles(IconButton)
        aria-label="Next"
        disabled={false}
        onClick={[Function]}
      >
        <pure(SkipNextIcon) />
      </WithStyles(IconButton)>
    </div>
    <div
      className="Controls-progressContainer-3"
    >
      <WithStyles(LinearProgress)
        classes={
          Object {
            "dashed": undefined,
            "root": "Controls-linearProgressRoot-4",
          }
        }
        value={0}
        variant="determinate"
      />
    </div>
    <MediaQuery
      query="(min-device-width: 768px)"
      values={Object {}}
    >
      <div
        className="Controls-metaContainer-5"
      >
        <img
          className="Controls-cover-9"
          role="presentation"
          src="https://some-host.com/muse-the-2nd-law.jpg"
        />
        <div
          className="Controls-infoContainer-6"
        >
          <h4
            className="Controls-songTitle-7"
          >
            Animals
          </h4>
          <h5
            className="Controls-songArtist-8"
          >
            Muse
          </h5>
        </div>
      </div>
    </MediaQuery>
  </div>
</div>
`;
